concepts/sepsis3/abx-micro-prescription.sql

```diff
@@ -0,0 +1,140 @@
+DROP TABLE IF EXISTS `physionet-data.mimic_derived.abx_micro_poe`;
+CREATE TABLE `physionet-data.mimic_derived.abx_micro_poe` as
+with abx as
+(
+  select pr.hadm_id
+  , pr.drug as antibiotic_name
+  , pr.starttime as antibiotic_time
+  , pr.stoptime as antibiotic_endtime
+  from `physionet-data.mimic_hosp.prescriptions` pr
+  -- inner join to subselect to only antibiotic prescriptions
+  inner join `physionet-data.mimic_derived.abx_poe_list` ab
+      on pr.drug = ab.drug
+)
+-- get cultures for each icustay
+-- note this duplicates prescriptions
+-- each ICU stay in the same hospitalization will get a copy of all prescriptions for that hospitalization
+, ab_tbl as
+(
+  select
+        ie.subject_id, ie.hadm_id, ie.stay_id
+      , ie.intime, ie.outtime
+      , abx.antibiotic_name
+      , abx.antibiotic_time
+      , abx.antibiotic_endtime
+  from `physionet-data.mimic_icu.icustays` ie
+  left join abx
+      on ie.hadm_id = abx.hadm_id
+)
+, me as
+(
+  select hadm_id
+    , chartdate, charttime
+    , spec_type_desc
+    , max(case when org_name is not null and org_name != '' then 1 else 0 end) as PositiveCulture
+  from `physionet-data.mimic_hosp.microbiologyevents`
+  group by hadm_id, chartdate, charttime, spec_type_desc
```

Comment:
instead of grouping by all this, group by the `micro_specimen_id`. aggregate the other columns

Reply:
What do you mean by aggregate other columns? Like use min/max/some other aggregate function? How would you know which one to use?

Reply:
yeah I just aggregated them, I usually do `MAX()` or `MIN()` - since all the values are the same, they will give the same result. It's kind of a quirk of SQL - you could group by as well, perhaps that's clearer.. chalk this one up to personal preference.